src/uct/ib/base/ib_iface.c

```diff
@@ -19,6 +19,14 @@
 #include <stdlib.h>
 #include <poll.h>
 
+#define UCT_IB_WORKER_AH_KEY 0x15599029u
+
+typedef struct uct_ib_ah_attr {
+    uct_worker_tl_data_t super;
+    struct ibv_ah_attr   attr;
+    struct ibv_pd        *pd;
+    struct ibv_ah        *ah;
```

Comment:
@hoopoepg what is the reason to put ah in worker tl data?

Reply:
re-implemented using hash